### PR TITLE
added new functionality to nxos_command

### PR DESCRIPTION
**Pull request by privateip**
- commands argument now accepts a dict arguments[1]
- waitfor has been renamed to wait_for with an alias to waitfor
- only show commands are allowd when check mode is specified
- config mode is no longer allowed in the command stack
- add argument match with valid values any, all

[1] The commands argument will now accept a dict argument that can
specifiy the output format of the command.  To specify a dict argument
use the form of { command: <str>, output: <str>, prompt: <str>,
response: <str> }.  Command and output are required arguments. Output
accepts valid values text and json.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_command.py [hunk 1]
```diff
@@ -36,8 +36,14 @@
         is returned.  If the I(waitfor) argument is provided, the
```
Comment (gundalow) [inline network/nxos/nxos_command.py]: `I(wait_for)`

We should standardise on the new name.

File: network/nxos/nxos_command.py [hunk 2]
```diff
@@ -46,6 +52,19 @@
...
+        values are C(all) or C(any).  If the value is set to C(all)
+        then all conditionals in the wait_for must be satisfied.  If
+        the value is set to C(any) then only one of the values must be
+        satisfied.
+    required: false
+    default: all
+    version_added: "2.2"
   retries:
     description:
       - Specifies the number of retries a command should by tried
```
Comment (gundalow) [inline network/nxos/nxos_command.py:70]: `I(wait_for)`

We should standardise on the new name.

Looks two lines below this, GitHub only lets you comment around the diff, not on any line.

**Comment by gundalow** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @privateip for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by privateip** [comment]
ready_for_review

**Comment by gundalow** [comment]
My comments haven't been addressed, nor have they been marked WONTFIX, was there meant to be another push?

**Comment by privateip** [comment]
Yeah forgot to commit my final changes 

ready_for_review

**Comment by gundalow** [comment]
shipit

**Pull request merged**